### PR TITLE
Set Minio user-agent for GCS calls

### PR DESCRIPTION
**Pull request by krishnasrinivas**
## Description

Send user-agent `Minio/2017-10-27T18:59:02Z (GPN:Minio;)`

## How Has This Been Tested?

added localhost as endpoint and saw that user-agent was as expected:
```
krishna@escape:~$ nc -l 9001
GET /b?alt=json&pageToken=&prefix=&project=miniotesting&projection=full HTTP/1.1
Host: localhost:9001
User-Agent: Minio/DEVELOPMENT.GOGET (GPN:Minio;)
Authorization: Bearer ya29.c.Elr-BFqo1Q-cMqqYCaTe7F3AoDGCR1SSUrBzra9DQQ2TpDZe0MrP7Iv6OL9deUdx4PGUw5CTWSwfs3r8jDElBKFDos3tlBZRVyBDXWzqlFRnK6CQ83H78Rz6Svs
X-Goog-Api-Client: gl-go/1.9.1 gccl/20170404
Accept-Encoding: gzip
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/gateway-gcs.go [hunk 1]
```diff
@@ -381,7 +382,7 @@ func newGCSGatewayLayer(projectID string) (GatewayLayer, error) {
 	}
 
 	// Initialize a GCS client.
-	client, err := storage.NewClient(ctx)
+	client, err := storage.NewClient(ctx, option.WithUserAgent(fmt.Sprintf("Minio/%s (GPN:Minio;)", Version)))
```
Comment (balamurugana) [inline cmd/gateway-gcs.go]: Is `GPN` google standard?

**Review by balamurugana — CHANGES_REQUESTED** [review]

File: cmd/gateway-gcs.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/gateway-gcs.go]: Only link i could find is this https://github.com/GoogleCloudPlatform/google-cloud-node/issues/2659#issuecomment-334867549 - GPN (Google Partner Network) is perhaps the terminology here.

**Review by harshavardhana — DISMISSED** [review]
LGTM tested.

File: cmd/gateway-gcs.go (hunk 1 above)
Comment (krishnasrinivas) [reply cmd/gateway-gcs.go]: yes @balamurugana this change was requested by them

**Review by balamurugana — DISMISSED** [review]

File: cmd/gateway-gcs.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/gateway-gcs.go]: perhaps its better to write a comment regarding this? @krishnasrinivas

**Review by harshavardhana — APPROVED** [review]
LGTM

**Review by balamurugana — APPROVED** [review]
LGTM

**Pull request merged**